pkg/machines/libvirt-dbus.js

```diff
@@ -541,6 +561,32 @@ LIBVIRT_DBUS_PROVIDER = {
         };
     },
 
+    /*
+     * Read properties of a single NodeDevice
+     *
+     * @param NodeDevice object path
+     */
+    GET_NODE_DEVICE({
+        id: objPath,
+        connectionName,
+    }) {
+        let props = {};
+
+        return dispatch => {
+            call(connectionName, objPath, 'org.libvirt.NodeDevice', 'GetXMLDesc', [0], TIMEOUT)
+                    .then(deviceXml => {
+                        let result = xml2js(deviceXml, { compact: true });
```

Comment:
Wrt. above, this is the bit that I'm cautious about: Is this really reversible, or can it be lossy?

Reply:
Removed xml2hs helper. Now if we need to change the existing XML we 'll use the domParser object itself.